hudi-client/src/test/java/org/apache/hudi/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -336,7 +335,54 @@ public void testDeletes() throws Exception {
   }
 
   /**
-   * Test scenario of new file-group getting added during upsert().
+   * Test update of a record to different partition with Global Index
+   */
+  @Test
+  public void testUpsertToDiffPartitionGlobaIndex() throws Exception {
```

Comment:
we should be able to test this functionality at the index level right? More test cases here keep adding to the test runtime... either way can we tst at the global index level?

Reply:
Have fixed one of the test for GlobaIndex as well.